plugin/storage/es/spanstore/reader.go

```diff
@@ -183,34 +203,34 @@ func (s *SpanReader) unmarshalJSONSpan(esSpanRaw *elastic.SearchHit) (*jModel.Sp
 }
 
 // Returns the array of indices that we need to query, based on query params
-func findIndices(startTime time.Time, endTime time.Time) []string {
+func findIndices(startTime time.Time, endTime time.Time, timeframe time.Duration) []string {
 	var indices []string
-	firstIndex := IndexWithDate(startTime)
-	currentIndex := IndexWithDate(endTime)
+	firstIndex := indexWithDate(startTime, timeframe)
+	currentIndex := indexWithDate(endTime, timeframe)
```

Comment:
why not make these indices member functions of the struct? Then you won't need to pass extra parameter which doesn't change anyway